datastore/sqlite.nim

```diff
@@ -172,9 +179,67 @@ proc sqlite3_column_text_not_null*(
 
   text
 
-template journalModePragmaStmt*(env: SQLite): RawStmtPtr =
+template pageSizePragmaStmt*(env: SQLite, pageSize: Positive): RawStmtPtr =
+  # https://www.sqlite.org/pragma.html#pragma_page_size
+  if not (isPowerOfTwo(pageSize) and pageSize >= 512 and pageSize <= 65536):
+    return failure "pageSize must be a power of two between 512 and 65536 inclusive"
+
+  let
+    s1 = prepare(env, "PRAGMA page_size = " & $pageSize & ";")
+
+  s1.dispose
+
+  let
+    s2 = prepare(env, "PRAGMA page_size;")
+
+  if (let x = sqlite3_step(s2); x != SQLITE_ROW):
+    s2.dispose
+    return failure $sqlite3_errstr(x)
```

Comment:
Probably not relevant to this PR, but how does a `Result[T, string]` equate to the return type of `RawStmtPtr`?

Reply:
It's a template, so gets expanded into the body of `proc new*(T: type SQLiteDatastore, ...): ?!T`, and in the case of `pageSizePragmaStmt(...)` the returned `RawStmtPtr` is assigned to a variable for which that type is the desired one.